EIPS/eip-867.md

```diff
@@ -0,0 +1,190 @@
+- **fromAddress** (*hex string*): The address from which ETH should be transferred
+- **toAddress** (*hex string*): The address to which ETH should be sent
+- **value** (*decimal string*): The amount of ETH to be transferred, in units of wei. The value must be a whole number greater than zero.
+
+
+#### storeCode
+The `storeCode` action stores the given code at the given address.
+
+*Data fields*
+
+- **type** (*string*): `storeCode`
+- **toAddress** (*hex string*): The address to which the contract should be restored.  
+- **expectedCodeHash** (*hex string*): the expected hash of the code already at the toAddress.The empty string should be used if no code is expected at the toAddress.  In all other cases, the “0x” prefix is optional.
+- **code** (*hex string*): the new bytecode associated with the contract
+
+### Appendix (Optional)
+The appendix can include additional supporting evidence or attachments that will help reviewers understand or verify the claims made in the ERP.  
+
+For the storeCode operation, it should include the proposed contract source (e.g. Solidity) as well as the other details required such that a reviewer can compile the source and generate the same bytecode.  It should also include the source that was originally stored at that address, if possible/applicable.  If the two contracts are not identical, changes should be noted. If they are identical, the author should indicate why no changes are necessary (this is unlikely). Additionally, any relevant reviews, audits, and test cases should be included to the extent that they address the issues encountered with the original contract.  
+
+If accepted, an ERP could easily compile the block at while the changes are to take place, and the source of the actions (which would be the output of the script, in a standardized object output). These can be bundled with the client for seamless execution.
+
+
+## ERP Approval Process
+Approval of any ERP will require a mandatory minimum of 30-days between the time the ERP is submitted (or substantially modified) and adoption by clients, and ERPs affecting more funds or more addresses may prompt a considerably longer review period.  In practice, ERPs will often require more time because the will likely be bundled into other client upgrades that are on their own schedule.  However, the 30-day mandatory review period  guarantees a period for comment and auditing and prevents any last minute inclusions from being rushed into an upcoming upgrade.  
+
+Once an ERP is published (via a pull request made to the EIP repository), the approval process begins.  The author is encouraged to seek feedback from the following groups:
+
+- People or organizations who are directly affected by the issue (e.g. entities who will be refunded or were associated with the original incident)
+- People or organizations that have a demonstrable history within the Ethereum community that are unaffected by the issue but may serve as relatively impartial adjudicators
```

Comment:
It seems like the EIP editors will need to judge who has a demonstrable history.

Reply:
I'm not sure of another way, although in cases where confirmation is straightforward, I suppose an EIP Editor could check the verification script themselves if it's less work.  "demonstrable history" was used to indicate that it's not enough to just have your neighbor create a github account and give a "thumbs up".  The goal is really just to set a high bar before engaging the EIP Editors and client teams.  Absolutely open to improvements on this part.

Reply:
When we did DAO "edge cases" administration, verifying the claims was one of the biggest problems. In the beginning, there was a spreadsheet where people were supposed to verify every line and mark them as "verified". After a while, we have realised that "people or organisations that have a demonstrable history within the Ethereum community..." are too busy and are not inclined to verify someone's recovery claims. So the spreadsheet had to rely on some "random" contributors. That, of course, did not work either, because no-one knew who actually verified anything or simply said that they did. Then we switched to the script writing, but in order to run the scripts, you would need to run parity or geth in tracing mode (which was a hassle back then, and would be prohibitive now). So even though I asked people to run the script, I doubt anyone actually did.

Sometimes we did get the people with demonstrable history to run some scripts and verify. One of these things happened just before extraBalance release. It took that person couple of days to be able to run the script, because, being such a person he was busy coding and had an unstable version of geth on his computer.

It is more work then you probably imagine. And back then, we did not need to convince a lot of people, we just needed 3 people to approve. I do not hold anything against your cause, but I am trying to convince you that this process is going to be a huge burden and will probably cost more in total than the amount of funds eventually returned.

Reply:
Fair enough, although I disagree with your last point:

> will probably cost more in total than the amount of funds eventually returned

There's nearly **half a billion USD** currently locked in multisigs alone.  1% of that could easily cover the effort required for ERPs for many years.  If the argument against is simply a resourcing problem, that's *easy* to solve.